### PR TITLE
refactor gui

### LINKED ISSUE #23: Rework GUI class
Currently the GUI class does a lot of different things, for now it is fine but I think we should try to limit this where possible. 

E.g. the show_popup function should be removed and that logic should be on the build menu instead. I think we could make use of the currently empty Popup.gd script if we want to implement a base class for all popups rather than passing everything into a function.

We should also create scripts for the other subcomponents of gui like the PowerMeter or Clock to manage their own functionality and state.

### PR DESCRIPTION
**Pull request by conalli**
overall aims
- move ui logic to individual component classes (clock, hydrogen meter etc)
- use signals to make ui generally more reactive (less need to call functions manually)
- decouple logic for managing resources from gui

closes #23

edit:
i also removed integer divisions giving warnings because i dont think they should be generated but what do you think?

### CODE REVIEW

File: project.godot [hunk 1]
```diff
@@ -17,8 +17,13 @@ config/icon="res://assets/sprites/flirty_pose.png"
 
 [autoload]
 
+StationEvent="*res://scripts/autoload/StationEvent.gd"
```
Comment (khatakslied) [inline project.godot:20]: So StationEvent is essentially the new play-through instance?

File: scripts/popup/GUIPopup.gd [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	label.text = text
+	return self
+	
+func connect_yes(f: Callable) -> GUIPopup:
+	yes.pressed.connect(f)
+	return self
+
+func connect_no(f: Callable) -> GUIPopup:
+	no.pressed.connect(f)
+	return self
```
Comment (khatakslied) [inline scripts/popup/GUIPopup.gd:22]: This is good. This is VERY good.

File: scripts/scene/Main.gd [hunk 3]
```diff
@@ -29,7 +19,7 @@ func _process(delta):
 	if delta_time >= 0.25:
 		delta_time = 0
 		update_in_game_time()
-		gui.update_clock(in_game_time)
+		Global.station.time = in_game_time
```
Comment (khatakslied) [inline scripts/scene/Main.gd:22]: Nice.

**Review by khatakslied — APPROVED** [review]
Fine work! Thank you.

File: project.godot (hunk 1 above)
Comment (conalli) [reply project.godot:20]: StationEvent is basically just a group of globally accessible signals that are emitted whenever the station variables are updated so things like the ui can be updated automatically

**Pull request merged**